File: mobile/lib/services/auth_service.dart
import 'package:dio/dio.dart';
import '../config/api_client.dart';

class AuthService {
  final ApiClient _apiClient = ApiClient();

  // Request OTP
  Future<Map<String, dynamic>> requestOtp({
    required String channel,
    required String destination,
    required String purpose,
  }) async {
    final response = await _apiClient.dio.post('/auth/otp/request', data: {
      'channel': channel,
      'destination': destination,
      'purpose': purpose,
    });
    return response.data;
  }

  // Verify OTP (with Firebase ID token for SMS, or OTP code for email)
  Future<Map<String, dynamic>> verifyOtp({
    required String channel,
    required String destination,
    String? idToken,
    String? otp,
    String? deviceId,
  }) async {
    final response = await _apiClient.dio.post('/auth/otp/verify', data: {
      'channel': channel,
      'destination': destination,
      if (idToken != null) 'idToken': idToken,
      if (otp != null) 'otp': otp,
      if (deviceId != null) 'deviceId': deviceId,
    });
    return response.data;
  }

  // Social login
  Future<Map<String, dynamic>> socialLogin({
    required String provider,
    required String idToken,
    String? deviceId,
  }) async {
    final response = await _apiClient.dio.post('/auth/social', data: {
      'provider': provider,
      'idToken': idToken,
      if (deviceId != null) 'deviceId': deviceId,
    });
    return response.data;
  }

  // Get current user
  Future<Map<String, dynamic>> getCurrentUser() async {
    final response = await _apiClient.dio.get('/users/me');
    return response.data;
  }

  // Refresh tokens
  Future<Map<String, dynamic>> refreshTokens(String refreshToken) async {
    final response = await _apiClient.dio.post('/auth/refresh', data: {
      'refreshToken': refreshToken,
    });
    return response.data;
  }
}

File: web/src/services/auth.service.ts
import apiClient from '../config/api';
import type {
  AuthResponse,
  RequestOtpRequest,
  VerifyOtpRequest,
  SocialLoginRequest,
  AuthTokens,
  User,
} from '../types/auth';

export const authService = {
  async requestOtp(data: RequestOtpRequest): Promise<{ message: string; otp?: string }> {
    const response = await apiClient.post('/auth/otp/request', data);
    return response.data;
  },

  async verifyOtp(data: VerifyOtpRequest & { idToken?: string }): Promise<AuthResponse> {
    const response = await apiClient.post('/auth/otp/verify', {
      ...data,
      // Backend expects idToken from Firebase, not plain OTP
      idToken: data.idToken,
    });
    return response.data;
  },

  async socialLogin(data: SocialLoginRequest): Promise<AuthResponse> {
    const response = await apiClient.post('/auth/social', data);
    return response.data;
  },

  async refreshTokens(refreshToken: string): Promise<AuthTokens> {
    const response = await apiClient.post('/auth/refresh', { refreshToken });
    return response.data.tokens || response.data;
  },

  async getCurrentUser(): Promise<{ user: User; roles: string[] }> {
    const response = await apiClient.get('/users/me');
    return response.data;
  },

  logout(): void {
    localStorage.removeItem('accessToken');
    localStorage.removeItem('refreshToken');
  },
};
